src/rgw/rgw_sal_motr.cc

```diff
@@ -1034,11 +1035,37 @@ int MotrBucket::list(const DoutPrefixProvider *dpp, ListParams& params, int max,
       rgw_bucket_dir_entry ent;
       auto iter = vals[i].cbegin();
       ent.decode(iter);
+      std::string key_name = ent.key.name + "/null";
+      if(keys[i] == key_name)
+      { 
+        // read data for null_index entry - (obj1/null)
+        bufferlist bl_null;
+        ret_code = this->store->do_idx_op_by_name(bucket_index_iname,
+                                    M0_IC_GET, key_name, bl_null);
+        ldpp_dout(dpp, 2) << "**** null key found for object -"<< keys[i] << dendl;
+        // read value for null index key(obj1/null)
+        ldpp_dout(dpp, 20) <<__func__<< "****** GET null rc : " << rc << dendl;
+        if (ret_code == 0)
+        {
+          bufferlist& blr = bl_null;
+          iter = blr.cbegin();
+          current_null_key.decode(iter);
+        }
+        ldpp_dout(dpp, 2) << "null key found : "<< keys[i] << dendl;
+        continue;
+      }
+      // obj1[null]
       if (params.list_versions || ent.is_visible())
+      {
+        if (ret_code == 0 && ent.key.instance == current_null_key.key.instance)
+          {
+            ldpp_dout(dpp, 2) << "**** inside if...making null to instance" << dendl;
+            ent.key.instance = "null";
```

Comment:
Why there is a need to explicitly assign "null" to instance in the list api, aren't we doing it during the put object itself. 
Let's discuss this.

Reply:
In put-object, we are generating and storing the version-id for all the obj versions(including the null version), But in the list-object-versions response for the null versions it should show the **'VersionID' ** field value as **'null'.**

Reply:
yes, removed the block. done